### DISCUSSION TITLE
[Question] Different extent in gempy model and depth map from GemGIS

### DISCUSSION THREAD

**Discussion by Changwan-planet** [cgre-aachen/gemgis#353] [Q&A]
Hi, @AlexanderJuestel

Your program plays a key role describing my data. Thank you for opening this program together with gempy!!!

I meet a obstacle to make a depth map or thickness map after creating gempy model.
It's working but showing the wrong thickness or depth in the graph **in the different extent**.

In the [tutorial](https://gemgis.readthedocs.io/en/latest/getting_started/tutorial/18_creating_depth_maps_from_gempy_models.html?highlight=depth) , 
```
{'Sand1': [PolyData (0x24a134c9fa0)
    N Cells:    4174
    N Points:   2303
    X Bounds:   9.720e+00, 9.623e+02
    Y Bounds:   1.881e+02, 9.491e+02
    Z Bounds:   3.050e+02, 7.250e+02
    N Arrays:   1,
  '#015482'],
 'Ton': [PolyData (0x24a137dfa00)
    N Cells:    5111
    N Points:   2739
    X Bounds:   9.720e+00, 9.623e+02
    Y Bounds:   3.578e+02, 1.058e+03
    Z Bounds:   3.050e+02, 7.265e+02
    N Arrays:   1,
  '#9f0052']} 
```

X, Y, Z bounds are related to the extent in the beginning of the code. 

But, When I print out the Polydata of my model, it' not related to the extent of the model.
the extent of my model  
```
extent = [216181., 216189., 342207., 342214., 140., 142.5]
```
but the Polydata shows
```
PolyData (0x7db787b46320)
  N Cells:    1213
  N Points:   657
  N Strips:   0
  X Bounds:   -2.481e-01, 2.500e-01
  Y Bounds:   -2.187e-01, 2.169e-01
  Z Bounds:   -8.014e-02, 5.800e-02
  N Arrays:   1
```

Why does the gempy model and depth map show different extent? 
It's too small compared with the extent. 

This is gempy model.
![Screenshot_from_2024-08-04_15-21-04.png](https://github.com/user-attachments/assets/222600e9-dd3a-47b7-b131-a6b911c6ece3)

They are depth map and thickness map.
![Screenshot_from_2024-07-31_20-06-08.png](https://github.com/user-attachments/assets/e806353f-5ae0-41c3-9cb2-58995d45ddb8)

![Screenshot_from_2024-08-02_17-15-21.png](https://github.com/user-attachments/assets/760be9f8-fcad-49d8-98ee-b4f9f35d6ad1)

I attached my python code, Points and Orientations to make gempy model. 

[Aug_4_quetsion_1.zip](https://github.com/user-attachments/files/16486541/Aug_4_quetsion_1.zip)

My goal is to export zmap like this [discussion](https://github.com/cgre-aachen/gemgis/discussions/321).

I hope you can help me solve my problem.
Thank you so much!!

**Comment by AlexanderJuestel** [comment]
@Changwan-planet, thanks for the question. This issue is related to how Gempy 3 works.

There should be a closed issue where I posted a work around, let me have a look

**Comment by AlexanderJuestel** [comment]
This should help you: https://github.com/gempy-project/gempy/issues/898#issuecomment-2242831847

**Reply by Changwan-planet** [reply]
OMG!
Very quick reply!!
I really appreciate your comment.
I will try that for my case!
Thanks:)

**Reply by Changwan-planet** [reply]
@AlexanderJuestel 

Thank you so much. I tried, but I got this error.

```
Traceback (most recent call last):
  File "/home/csun/GEMGIS/Miho/Aug_1_Depthmap/test.py", line 147, in <module>
    vertices = geo_data.input_transform.apply_inverse(geo_data.solutions.raw_arrays.vertices[index])
AttributeError: 'GeoModel' object has no attribute 'input_transform'. Did you mean: 'update_transform'?
```
 Is this also related to the gempy version control? 

==========================================================================
Updated comment

Ok. I am trying with ```transform```. I will update the result.

**Reply by Changwan-planet** [reply]
@AlexanderJuestel 
Your suggestion was successful!!

I used .transform.
and
np.flipud(z_values) to visualize and export.
I share my neat results!!!

1.Depth map with coordinate that I used when I created gempy model 
![Screenshot_from_2024-08-04_20-20-46.png](https://github.com/user-attachments/assets/bec649b8-6ced-4f84-8749-466ec1b2a585)

2.Zmap from gempy model based on this discussion.
![Figure_1.png](https://github.com/user-attachments/assets/01d1754d-b651-4ee0-9816-b694297e8724)

3. In QGIS, Hillshade of zmap overlayed by itself with points when I used to make gempy model.
![Miho_gempy_model_hillshade+dem+points.png](https://github.com/user-attachments/assets/9cdcff96-6498-4b1c-a0c2-e6d3d6027877)


Thank you so much 😊
